### PR TITLE
Storage: minor refine of bitmap and add more tests

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### What problem does this PR solve?

Issue Number: ref #6296 

Problem Summary:

### What is changed and how it works?

1. try to merge continuous ranges when build bitmap.
2. add simd flag to `BitmapFilter.cpp` and `DMVersionFilterBlockInputStream.cpp`
3. add more tests 

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/DeltaMerge/Segment.cpp [hunk 1]
```diff
@@ -2295,79 +2294,99 @@ BitmapFilterPtr Segment::buildBitmapFilterNormal(const DMContext & dm_context,
...
+                some_packs_set->insert(pack_id);
+                continue;
+            }
 
-        if (pack_stat.not_clean > 0)
-        {
-            // We need to read this pack to do MVCC filter.
-            some_packs->insert(pack_id);
-            continue;
+            if (offset + rows == preceded_rows - pack_stat.rows)
```
Comment (hongyunyan) [inline dbms/src/Storages/DeltaMerge/Segment.cpp:2368]: It's better to add some comments here.

File: dbms/src/Storages/DeltaMerge/Segment.cpp [hunk 2]
```diff
@@ -2377,44 +2396,33 @@ BitmapFilterPtr Segment::buildBitmapFilterStableOnly(const DMContext & dm_contex
...
-                                                                    dm_context,
-                                                                    read_ranges,
-                                                                    filter,
-                                                                    max_version);
-        all_match = all_match && (all_packs[i].size() == dmfiles[i]->getPacks());
-        some_packs_count += some_packs[i]->size();
-    }
+    RUNTIME_CHECK(!dmfiles.empty());
+
+    auto [all_packs, some_packs] = parseDMFilePackInfo(dmfiles, dm_context, read_ranges, filter, max_version);
```
Comment (hongyunyan) [inline dbms/src/Storages/DeltaMerge/Segment.cpp:2401]: may be we can find a better name for "all_packs and some_packs"?

Comment (Lloyd-Pottiger) [reply dbms/src/Storages/DeltaMerge/Segment.cpp]: done

File: dbms/src/Storages/DeltaMerge/Segment.cpp (hunk 1 above)
Comment (Lloyd-Pottiger) [reply dbms/src/Storages/DeltaMerge/Segment.cpp:2370]: done

**Review by hongyunyan — APPROVED** [review]
LGTM

File: dbms/src/Storages/DeltaMerge/Segment.cpp [hunk 3]
```diff
@@ -2295,79 +2294,101 @@ BitmapFilterPtr Segment::buildBitmapFilterNormal(const DMContext & dm_context,
...
+            {
+                continue;
+            }
 
-        // assert(handle_res[pack_id] == RSResult::All);
+            if (handle_res[pack_id] == RSResult::Some
+                || pack_stat.not_clean > 0
+                || pack_filter.getMaxVersion(pack_id) > max_version)
+            {
+                // We need to read this pack to do RowKey filter.
```
[the remark this answers is not in the document]
Comment (Lloyd-Pottiger) [reply dbms/src/Storages/DeltaMerge/Segment.cpp]: done

**Review by hehechen — APPROVED** [review]
Do we have UT with continues `all_packs` ranges?

File: dbms/src/Storages/DeltaMerge/Segment.cpp [hunk 4]
```diff
@@ -2295,79 +2294,101 @@ BitmapFilterPtr Segment::buildBitmapFilterNormal(const DMContext & dm_context,
...
+    std::vector<Range> skipped_ranges;
     // Packs that some rows compliant with MVCC filter and RowKey filter requirements.
     // We need to read these packs and do RowKey filter and MVCC filter for them.
-    auto some_packs = std::make_shared<IdSet>();
-    UInt64 rows = 0;
-    for (size_t pack_id = 0; pack_id < pack_stats.size(); pack_id++)
+    std::vector<IdSetPtr> some_packs_sets;
+    some_packs_sets.reserve(dmfiles.size());
+
+    size_t offset = 0;
```
[the remark this answers is not in the document]
Comment (Lloyd-Pottiger) [reply dbms/src/Storages/DeltaMerge/Segment.cpp:2328]: done

**Comment by Lloyd-Pottiger** [comment]
> Do we have UT with continues `all_packs` ranges?

yes, `SegmentBitmapFilterTest.CleanStable`

**Comment by Lloyd-Pottiger** [comment]
> Do we have UT with continues `all_packs` ranges?

added.

File: dbms/src/Storages/DeltaMerge/Segment.cpp [hunk 5]
```diff
@@ -2377,44 +2398,37 @@ BitmapFilterPtr Segment::buildBitmapFilterStableOnly(const DMContext & dm_contex
...
+    if (skipped_ranges.size() == 1 && skipped_ranges[0].offset == 0 && skipped_ranges[0].rows == segment_snap->stable->getDMFilesRows())
     {
         LOG_DEBUG(log, "all match, total_rows={}, cost={}ms", segment_snap->stable->getDMFilesRows(), sw.elapsedMilliseconds());
         return std::make_shared<BitmapFilter>(segment_snap->stable->getDMFilesRows(), /*default_value*/ true);
     }
 
     auto bitmap_filter = std::make_shared<BitmapFilter>(segment_snap->stable->getDMFilesRows(), /*default_value*/ false);
-    UInt32 preceded_dmfile_rows = 0;
-    for (size_t i = 0; i < dmfiles.size(); i++)
+    for (const auto & pack : skipped_ranges)
```
[the remark this answers is not in the document]
Comment (Lloyd-Pottiger) [reply dbms/src/Storages/DeltaMerge/Segment.cpp]: fixed

**Pull request merged**